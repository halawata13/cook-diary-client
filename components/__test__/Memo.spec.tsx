import React from 'react';
import renderer, { act } from 'react-test-renderer';
import Memo from '../Memo';

describe('Memo', () => {
  let component: renderer.ReactTestRenderer;
  let tree: renderer.ReactTestRendererJSON | renderer.ReactTestRendererJSON[] | null;

  it('should match to snapshot', () => {
    act(() => {
      component = renderer.create(<Memo memo={''} onMemoChanged={() => {}} editable={true} />);
      tree = component.toJSON();
    });

    expect(tree).toMatchSnapshot();
  });

  // it('should match to memo props', async () => {
  //   act(() => {
  //     const component = renderer.create(<Memo memo={'memomemo'} onMemoChanged={value => value} editable={true} />);
  //     const textarea = component.root.findByProps({ id: 'memo' })
  //     expect(textarea.props.value).toStrictEqual('memomemo');
  //     expect(textarea.props.readOnly).toBeFalsy();
  //
  //     textarea.props.onChange({ currentTarget: { value: 'mome' } });
  //     expect(textarea.props.value).toStrictEqual('mome');
  //   });
  // });
});
