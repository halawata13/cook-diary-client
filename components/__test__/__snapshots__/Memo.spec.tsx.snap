// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Memo should match to snapshot 1`] = `null`;
